### PR TITLE
feat: [CDS-77646]: Exposed Execute pipeline V1 spec endpoint And Fixed create pipeline V1 spec API flow for v1 yaml

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes
- Exposed v1 spec execute API for yaml v1 version
- Fixed v1 spec pipeline create API for yaml v1 version

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51986)

### CODE REVIEW

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 2 of 4 files at r1, 1 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51986)* status: 1 file(s) to review.

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -704,12 +724,72 @@ paths:
       tags:
         - Schemas
       x-internal: true
+      x-stoplight:
+        id: n2nrw5l7mja9b
+  '/v1/orgs/{org}/projects/{project}/pipeline/{pipeline}/execute':
+    parameters:
+      - schema:
+          type: string
+        name: org
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:733]: Please use the common parameters defined for these (refer to other APIs)

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -704,12 +724,72 @@ paths:
...
+            default: 'false'
+          in: query
+          name: notify_only_user
+        - schema:
+            type: string
+          in: query
+          name: notes
+        - $ref: '#/components/parameters/AccountHeader'
+      tags:
+        - Pipelines
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:786]: nit - This tag can be "Pipeline Execute" also if we want to add more Execution related apis in a new Tag.

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -2472,6 +2554,28 @@ components:
...
+      properties:
+        executionDetails:
+          $ref: '#/components/schemas/ExecutionDetails'
+    ExecutionDetails:
+      title: ExecutionDetails
+      x-stoplight:
+        id: qshycktu12bju
+      type: object
+      properties:
+        executionId:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:2571]: casing should be snake case

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 4]
```diff
@@ -3274,6 +3378,16 @@ components:
         application/json:
           schema:
             $ref: '#/components/schemas/IndividualSchemaResponseBody'
+    PipelineExecuteResponseBody:
+      description: Example response
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/PipelineExecuteResponseBody'
+        application/xml:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:3387]: is xml needed? better to add support for yaml

**Review by mankrit-singh — COMMENTED** [review]
added few comments

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 6 files at r4, 8 of 8 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51986)* status: 9 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 3387 at r2](https://reviewable.io/reviews/harness/harness-core/51986#gh-1306899776:-Nd03abD74AOznec8uyH:b-myofjm) ([raw file](https://github.com/harness/harness-core/blob/01d587dd39f0b6e2ca5f2d6c95ba1810ba08e265/pipeline-service/contracts/openapi/v1/openapi.yaml#L3387)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

is xml needed? better to add support for yaml
</blockquote></details>

no yet

**Review by archit-harness — APPROVED** [review]
> Reviewed 3 of 4 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51986)* status: 5 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by utkarshchoubey-harness** [comment]
trigger uts

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 5]
```diff
@@ -704,12 +724,59 @@ paths:
...
+      responses:
+        '200':
+          $ref: '#/components/responses/PipelineExecuteResponseBody'
+      x-stoplight:
+        id: 15a980rok3f3a
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
```
[the remark this answers is not in the document]
Comment (utkarshchoubey-harness) [reply pipeline-service/contracts/openapi/v1/openapi.yaml:744]: will confirm and raise a separate pr if applicable

**Review by utkarshchoubey-harness — APPROVED** [review]

**Comment by utkarshchoubey-harness** [comment]
trigger ut0

**Comment by utkarshchoubey-harness** [comment]
trigger ut1

**Comment by utkarshchoubey-harness** [comment]
trigger ut2

**Comment by utkarshchoubey-harness** [comment]
trigger ut4

**Comment by utkarshchoubey-harness** [comment]
trigger review

**Comment by utkarshchoubey-harness** [comment]
trigger ut0

**Comment by utkarshchoubey-harness** [comment]
trigger ut1

**Comment by utkarshchoubey-harness** [comment]
trigger ut2

**Comment by utkarshchoubey-harness** [comment]
trigger ut3

**Comment by utkarshchoubey-harness** [comment]
trigger ut4

**Comment by utkarshchoubey-harness** [comment]
trigger ut5

**Pull request merged**